pkgs/top-level/all-packages.nix

```diff
@@ -13387,8 +13390,12 @@ let
 
   vym = callPackage ../applications/misc/vym { };
 
-  w3m = callPackage ../applications/networking/browsers/w3m {
+  w3m = callPackage ../applications/networking/browsers/w3m { };
+
+  w3m-pure = w3m.override {
     graphicsSupport = false;
+    x11Support = false;
+    mouseSupport = false;
```

Comment:
Yep, it is more consistent, consistent is good, I can do that. But even
then I'd still like to have an explicitly minimal version in case
someone else will fiddle with defaults in the future. `w3m-pure` is for
derivations that use w3m in batch mode for text rendering, not browsing,
and I'd like that information to be explicit in nixpkgs. Which means
there would be three `w3m`'s: `w3m`, `w3mFull`, `w3mMinimal` (current
`w3m-pure`) and

* I have no idea what should be the difference between `w3m` and
  `w3mFull` if the default derivation should be what "most users will want
  by default".
* ranger will have to use `w3mFull` or specifically crafted version (4th
  `w3m`!) with image support.

So, should I rename `w3m-pure` to `w3mMinimal` and define `w3mFull =
w3m`?

(Btw, I still think that using nixpkgs.config options (constantly
rejected in nixpkgs) is a good thing as it simplifies things a lot, e.g.
it can simplify these four `w3m`s into two and one assert in `ranger`,
but whatever.)